plugins/filter/subkey.py

```diff
@@ -7,31 +7,39 @@
 import json
 
 
-def subkey_inspect(a, network='', scheme='', public=False):
+def subkey_inspect(uri, network='', scheme='', public=False):
     """Run subkey inspect command and return output."""
-    a = to_text(a, errors='surrogate_or_strict', nonstring='simplerepr')
+    uri = to_text(uri, errors='surrogate_or_strict', nonstring='simplerepr')
 
     args = []
+    log_uri = '<URI>'
     if scheme:
         args.extend(['--scheme', scheme])
     if network:
         args.extend(['--network', network])
     if public:
         args.extend(['--public'])
+        log_uri = uri
 
     try:
-        process = subprocess.Popen(['subkey', 'inspect', a, "--output-type=json", *args],
+        process = subprocess.Popen(['subkey', 'inspect', uri, "--output-type=json", *args],
                                    stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE,
                                    universal_newlines=True)
+    except FileNotFoundError:
+        raise AnsibleFilterError(
+            "subkey binary is required for this filter. Please, install it on local machine: sudo curl -fSL -o "
+            "/usr/local/bin/subkey 'https://releases.parity.io/substrate/x86_64-debian%3Astretch/v3.0.0/subkey/subkey' "
```

Comment:
is there a reason for not including its installation within this role?

Reply:
that makes sense, though why not run it on the node then?

Reply:
There are two reasons why this is filter and not part of the ansible role:
1. It was planned to be used in multiple roles.
2. Filter executed in check mode. 

Why it was developed as an ansible filter and not as an ansible module (modules are executed on a remote machine)
1. We will have an extra task if we use module.  For example: 1. inspect key task.  2. task which uses the key.  With the filter, we need only a second task.
2. Filer is more suitable since we just manipulate data. 
3. models use the python library as a dependency, not the binary (should be cross-platform). But since [py-substrate-interface](https://github.com/polkascan/py-substrate-interface) is available we can rewrite it as a module.